rcl/fibonacci_action_server/main.c

```diff
@@ -0,0 +1,234 @@
+
+    *fib_args->feedback_lenght = i;
+    fib_args->feedback[i] = fib_args->feedback[i-1] + fib_args->feedback[i-2];
+    
+    usleep(500000);
+  }
+
+  *fib_args->goal_done = true;
+}
+
+int main(int argc, const char * const * argv)
+{
+  rcl_ret_t rv;
+
+  rcl_init_options_t options = rcl_get_zero_initialized_init_options();
+  rv = rcl_init_options_init(&options, rcl_get_default_allocator());
+  if (RCL_RET_OK != rv) {
+    printf("rcl init options error: %s\n", rcl_get_error_string().str);
+    return 1;
+  }
+
+  rcl_context_t context = rcl_get_zero_initialized_context();
+  rv = rcl_init(argc, argv, &options, &context);
+  if (RCL_RET_OK != rv) {
+    printf("rcl initialization error: %s\n", rcl_get_error_string().str);
+    return 1;
+  }
+
+  rcl_node_options_t node_ops = rcl_node_get_default_options();
+  rcl_node_t node = rcl_get_zero_initialized_node();
+  rv = rcl_node_init(&node, "fibonacci_action_server_rcl", "", &context, &node_ops);
+  if (RCL_RET_OK != rv) {
+    printf("Node initialization error: %s\n", rcl_get_error_string().str);
+    return 1;
+  }
+
+  const char * action_name = "fibonacci";
+  const rosidl_action_type_support_t * action_type_support = ROSIDL_GET_ACTION_TYPE_SUPPORT(example_interfaces, Fibonacci);
+  rcl_action_server_t action_server = rcl_action_get_zero_initialized_server();
+  rcl_action_server_options_t action_server_ops = rcl_action_server_get_default_options();
+
+  rcl_clock_t clock;
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  rv = rcl_ros_clock_init(&clock, &allocator);
+
+  rv = rcl_action_server_init(
+      &action_server, 
+      &node, 
+      &clock,
+      action_type_support, 
+      action_name, 
+      &action_server_ops);
+
+  if (RCL_RET_OK != rv) {
+    printf("Action server initialization error: %s\n", rcl_get_error_string().str);
+    return 1;
+  }
+
+  size_t num_subscriptions, num_guard_conditions, num_timers, num_clients, num_services;
+
+  rcl_action_server_wait_set_get_num_entities(&action_server, &num_subscriptions, &num_guard_conditions, &num_timers, &num_clients, &num_services);
+
+  rcl_wait_set_t wait_set = rcl_get_zero_initialized_wait_set();
+  rv = rcl_wait_set_init(&wait_set, num_subscriptions, num_guard_conditions, num_timers, num_clients, num_services, 0, &context, rcl_get_default_allocator());
+  if (RCL_RET_OK != rv) {
+    printf("Wait set initialization error: %s\n", rcl_get_error_string().str);
+    return 1;
+  }
+
+  rcl_action_goal_handle_t * goal_handle;
+  rcl_action_goal_info_t goal_info;
+  pthread_t goal_thread;
+
+  bool processing_goal = false;
+  bool goal_done = false;
+  int32_t * feedback;
+  int32_t feedback_lenght;
+  int32_t goal_order;
+
+  do {
+    rv = rcl_wait_set_clear(&wait_set);
+    if (RCL_RET_OK != rv) {
+      printf("Wait set clear error: %s\n", rcl_get_error_string().str);
+      return 1;
+    }
+    
+    size_t index;
+    rv = rcl_action_wait_set_add_action_server(&wait_set, &action_server, &index);
```

Comment:
There are many functions that return a `rcl_ret_v`, some of then are checked while other are not, why?

Reply:
Fixed